_inc/client/components/upgrade-notice-content/index.jsx

```diff
@@ -82,33 +105,39 @@ const UpgradeNoticeContent = moduleSettingsForm(
 				// Track the main toggle switch.
 				analytics.tracks.recordJetpackClick( {
 					target: 'jetpack_site_accelerator_toggle',
-					toggled: 'on'
+					toggled: 'on',
+					from: 'warm_welcome'
 				} );
 			} else {
 				analytics.tracks.recordJetpackClick( {
 					target: 'jetpack_site_accelerator_toggle',
-					toggled: 'off'
+					toggled: 'off',
+					from: 'warm_welcome'
 				} );
 			}
 
 			// Track any potential Photon toggle switch.
 			if ( this.props.getOptionValue( 'photon' ) !== newPhotonStatus ) {
 				analytics.tracks.recordEvent( 'jetpack_wpa_module_toggle', {
 					module: 'photon',
-					toggled: ( false === newPhotonStatus ) ? 'off' : 'on'
+					toggled: ( false === newPhotonStatus ) ? 'off' : 'on',
+					from: 'warm_welcome'
 				} );
 			}
 
 			// Track any potential Photon CDN toggle switch.
 			if ( this.props.getOptionValue( 'photon-cdn' ) !== newAssetCdnStatus ) {
 				analytics.tracks.recordEvent( 'jetpack_wpa_module_toggle', {
 					module: 'photon-cdn',
-					toggled: ( false === newAssetCdnStatus ) ? 'off' : 'on'
+					toggled: ( false === newAssetCdnStatus ) ? 'off' : 'on',
+					from: 'warm_welcome'
 				} );
 			}
 		};
 
 		renderInnerContent() {
+			this.trackNoticeView();
```

Comment:
This seems to be triggered multiple times when the page loads, and never includes the version:

<img width="811" alt="screenshot 2018-11-02 at 11 23 33" src="https://user-images.githubusercontent.com/426388/47910019-d3cf4300-de91-11e8-9209-6361dc5da771.png">

Should we have the track event somewhere else instead, like on `componentDidMount`?

Reply:
`componentDidMount` yes!

Reply:
Done!